address/address.c

```diff
@@ -44,8 +44,22 @@ const char AddressSpecials[] = "@.,:;<>[]\\\"()";
 /**
  * is_special - Is this character special to an email address?
  * @param ch Character
+ * @param mask Bitmask of characters 32-91 that are special (others are always zero)
  */
-#define is_special(ch) strchr(AddressSpecials, ch)
+#define is_special(ch, mask) ((ch) >= 32 && (ch) < 96 && ((mask >> ((ch) - 32)) & 1))
+
+/* AddressSpecials, for is_special() */
+#define ADDRESS_SPECIAL_MASK 0x380000015c005304ULL
+
+/* AddressSpecials except . " ( \ */
+#define USER_SPECIAL_MASK 0x280000015c001200ULL
+
+/* AddressSpecials except . ( [ ] \ */
+#define DOMAIN_SPECIAL_MASK 0x000000015c001204ULL
+
+/* AddressSpecials except , . \ [ ] ( */
+#define ROUTE_SPECIAL_MASK 0x0000000058000204ULL
```

Comment:
I wrote a small script to make these, it's entirely possible I messed up (I combined with some manual work). I'll recalc and see.

Reply:
Re-did the script, and it comes out to your value. I also sorted the comments by ASCII value so that it's slightly easier to compare the exception masks.